File: keyword.js
var https = require('https');
const crypto = require('crypto');
var sha256 = require('js-sha256');
var request = require('request');

const http_method = 'GET';
const request_uri = '/keywordstool';
const CUSTOMER_ID  = '1125310';
const API_KEY = '010000000047af83ccb256d12031c1e842a87e3c1bb9a4878096687d53c978fa08f6a7534f'
const API_SECRET = 'AQAAAABHr4PMslbRIDHB6EKofjwb7cIBG9AJjKeSNlKqlJJAGQ==';
var TIMESTAMP = new Date().getTime();

//서명생성
const temp = TIMESTAMP+'.'+http_method+'.'+request_uri;
var SIGNATURE = crypto.createHmac('sha256', API_SECRET).update(temp).digest('hex');
console.log(SIGNATURE);

SIGNATURE = sha256.hmac(API_SECRET, TIMESTAMP + '.GET./keywordstool');
console.log(SIGNATURE);

//헤더 설정
var options = {
  host : 'api.naver.com',
//    url: 'https://api.naver.com/keywordstool',
    path: request_uri,
    method: http_method,
    headers: {
      'X-Timestamp' : TIMESTAMP,
      'X-API-KEY':API_KEY,
      'X-Customer': CUSTOMER_ID,
      'X-Signature' : SIGNATURE,
      'Content-Type': 'application/json'
    }
};

// request.get(options, function (error, response, body) {
//    console.log("statusCode: ", response.statusCode);
//   console.log('HEADERS: ' + JSON.stringify(response.headers));
// //  if (!error && response.statusCode == 200) {
//   //  var objBody = JSON.parse(response.body);
// //    console.log('object : ', response.caseless.dict['content-length']);
// //
// //  }
// });
//


var req = https.request(options, function(res) {
  console.log("statusCode: ", res.statusCode);
  console.log('HEADERS: ' + JSON.stringify(res.headers));
      res.setEncoding('utf8');
      res.on('data', function (xml) {
          console.log(xml);
      });
  });
  req.end();

File: keyword.py
#-*- coding: utf-8 -*-
import pycurl
import hmac, hashlib, base64, time, urllib, json, sys
from io import BytesIO

def getSignature(secret, method, path):
    message = str(int(time.time())) + "." + method + "." + path
    dig = hmac.new(secret, msg=bytearray(message, 'utf-8'), digestmod=hashlib.sha256).digest()
    sig = base64.b64encode(dig)
    return sig

baseURL = "https://api.naver.com"
apiSecret = b"AQAAAABHr4PMslbRIDHB6EKofjwb7cIBG9AJjKeSNlKqlJJAGQ=="
apiKey = "010000000047af83ccb256d12031c1e842a87e3c1bb9a4878096687d53c978fa08f6a7534f"
apiCustomerID = "1125310"

signature = getSignature(apiSecret, "GET", "/keywordstool")

buf = BytesIO()
# c = pycurl.Curl()
# c.setopt(pycurl.SSL_VERIFYPEER, 0)
# c.setopt(pycurl.SSL_VERIFYHOST, 0)
# c.setopt(c.WRITEDATA, buf)
# url = baseURL + "/keywordstool?hintKeywords=" + urllib.quote(sys.argv[1])
# c.setopt(c.URL, url)
# c.setopt(pycurl.HTTPHEADER, ['X-API-KEY: '+ apiKey,
#                                    'X-Customer: ' + apiCustomerID,
#                                    'X-Timestamp: ' + str(int(time.time())),
#                                    'X-Signature: ' + signature])
# c.perform()
#
# body = buf.getvalue()
#
# data = json.loads(body)
#
#
# keyword = data["keywordList"][0]["relKeyword"]
# PC = data["keywordList"][0]["monthlyPcQcCnt"]
# mobile = data["keywordList"][0]["monthlyMobileQcCnt"]
# Total = PC+mobile

blog = 1
PC =2
mobile =3
Total = 4

#print(cListByIds)
print ("[%s]의 검색량 입니다!\n블로그 문서량 : %d\n모바일 검색량 : %d\nPC검색량 : %d\nToTal 검색량 : %d\n" % (sys.argv[1],blog,mobile,PC,Total))
